docs/core/docker/build-container.md

```diff
@@ -0,0 +1,391 @@
+---
+title: Containerize an app with Docker tutorial
+description: In this tutorial, you will learn how to containerize a .NET Core application with Docker.
+ms.date: 03/20/2019
+ms.topic: tutorial
+ms.custom: "mvc, seodec18"
+#Customer intent: As a developer, I want to containerize my .NET Core app so that I can deploy it to the cloud.
+---
+
+# Tutorial: Containerize a .NET Core app
+
+This tutorial teaches you how to build a Docker image that contains your .NET Core application. The image can be used to create containers for your local development environment, private cloud, or public cloud.
+
+In this tutorial you'll learn how to:
+
+> [!div class="checklist"]
+> * Create and publish a simple .NET Core app
+> * Create and configure a Dockerfile for .NET Core
+> * Build a Docker image
+> * Create and run a Docker container
+
+This tutorial teaches the Docker container build and deploy tasks for a .NET Core application. The *Docker platform* uses the *Docker engine* to quickly build and package apps as *Docker images*. These images are written in the *Dockerfile* format to be deployed and run in a layered container.
+
+## Prerequisites
+
+To complete this tutorial, install the following prerequisites:
+
+- [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download)\
+If you have .NET Core installed, use the `dotnet --info` command to determine which SDK you're using.
+
+- [Docker Community Edition](https://www.docker.com/products/docker-desktop)
+
+- A temporary working directory for the *Dockerfile* and .NET Core example app.
+
+## Create .NET Core app
+
+You need a .NET Core app that the Docker container will run. Open your terminal, create a working directory, and enter it. In the working directory, run the following command to create a new project in a subdirectory named app:
+
+```console
+dotnet new console -o app -n myapp
+```
+
+That command creates a new directory named *app* and generates a "Hello World" app. You can test this app to see what it does. Enter the *app* directory and execute the command `dotnet run`. You'll see the following output:
+
+```console
+> dotnet run
+Hello World!
+```
+
+The default template creates an app that prints to the terminal and then exits. For this tutorial, you'll use an app that loops indefinitely. Open the **Program.cs** file in a text editor. It should currently look like the following code:
+
+```csharp
+using System;
+
+namespace myapp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
+```
+
+Replace the file with the following code that counts numbers every second:
+
+```csharp
+using System;
+
+namespace myapp
+{
+    class Program
+    {
+        static void Main(string[] args)
```

Comment:
This feature is from 7.1, so should be available for any user.

```suggestion
        static async Task Main(string[] args)
```

Reply:
the default template for 2.2 seems to target 7.0 as the language :(

Reply:
Yes, but you can set it with this in your csproj:

```XML
<LangVersion>latest</LangVersion>
```

And, that will be changing with the next .NET Core SDK (where latest or preview is the default).  See here: https://docs.microsoft.com/dotnet/csharp/language-reference/configure-language-version

Reply:
Is there a `dotnet` command to do this? I would rather not have to go into more steps about the csproj file. That just adds more stuff for the user to read about the sample app when the intent of the article is about getting the app into the docker container.

Reply:
Not that I know about. You can ignore this suggestion, if you prefer. I agree it's not the focus.